web/yo/app/scripts/controllers/gene.js

```diff
@@ -290,12 +289,18 @@ angular.module('oncokbApp')
             function mostRecentItem(reviewObjs) {
                 var mostRecent = -1;
                 for (var i = 0; i < reviewObjs.length; i++) {
-                    if (mostRecent < 0) {
-                        if (reviewObjs[i].get('updateTime')) {
+                    var currentItemTime = new Date(reviewObjs[i].get('updateTime'));
+                    // we only continue to check if current item time is valid
+                    if (currentItemTime instanceof Date && !isNaN(currentItemTime.getTime())) {
+                        if (mostRecent < 0) {
                             mostRecent = i;
+                        } else {
+                            // reset mostRect time when current item time is closer
+                            var mostRecentTime = new Date(reviewObjs[mostRecent].get('updateTime'));
```

Comment:
Can we reuse the code you wrong before for validating the Date? And `mostRecentTime` needs to be checked as well.